crates/router/src/core/payments.rs

```diff
@@ -356,9 +356,11 @@ where
         .add_access_token(state, &connector, merchant_account)
         .await?;
 
-    match add_access_token_result.access_token_result {
-        Ok(access_token) => router_data.access_token = access_token,
-        Err(connector_error) => router_data.response = Err(connector_error),
+    if add_access_token_result.connector_supports_access_token {
```

Comment:
can we make this into a function, we have the same pattern in 3 different places, will this guard be required in those other places?